packages/storage/src/file.js

```diff
@@ -1785,6 +1795,55 @@ File.prototype.save = function(data, options, callback) {
 };
 
 /**
+ * Set the storage class for this file.
+ *
+ * @resource [Per-Object Storage Class]{@link https://cloud.google.com/storage/docs/per-object-storage-class}
+ * @resource [Storage Classes]{@link https://cloud.google.com/storage/docs/storage-classes}
+ *
+ * @param {string} storageClass - The new storage class. (`multi_regional`,
+ *     `regional`, `nearline`, `coldline`)
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ *
+ * @example
+ * file.setStorageClass('regional', function(err, apiResponse) {
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   // The storage class was updated successfully.
+ * });
+ *
+ * //-
+ * // If the callback is omitted, we'll return a Promise.
+ * //-
+ * file.setStorageClass('regional').then(function() {});
+ */
+File.prototype.setStorageClass = function(storageClass, callback) {
+  var self = this;
+
+  // In case we get input like `storageClass`, convert to `storage_class`.
+  storageClass = storageClass
+    .replace(/-/g, '_')
+    .replace(/([a-z])([A-Z])/g, function(_, low, up) {
+      return low + '_' + up.toLowerCase();
+    });
```

Comment:
The upstream API allows lower case, the camelcase -> snakecase is just a helper we have in case the user couldn't remember. However, I've changed this to `toUpperCase()` all the input that we get.